bootloader/src/pyi_archive.c

```diff
@@ -235,14 +235,15 @@ pyi_arch_extract2fs(ARCHIVE_STATUS *status, TOC *ptoc)
  * The search space uses the given sizes because on Windows and OS X, the code signing
  * will add padding between the end of the COOKIE and the beginning of the signature
  * to align the signature to a quadword or a page boundary respectively. On Linux,
- * we use objtool to insert the archive into the bootloader, and objtool will
- * move the ELF section headers so they follow the cookie, so we need to search backward
- * past the section headers to find the cookie.
+ * we use objtool to insert the archive into the bootloader and on runtime parse the
+ * ELF headers to extract the precise end offset.
```

Comment:
The backward search is still in for non ELF files, isn't it? Thus the comment should state this.

Reply:
@htgoebel Going through the feedback here today I have come to the understanding that other Unixes might append the payload at the end even if their file format is also ELF. If there is no fixup or signing as on MachO binaries on OSX, is a backward search necessary on other Unixes or is the data right there at the end?